NOAA-EMC/ufsatm
# Add 2d domain decomposition to the inline post

## Description

Resubmit an update from previous PR, after many updates in upp and other fv3atm submodules.

The community upp is currently decomposed on latitudes (y-direction) only. As the ufs-weather-model is making 2D decomposition in both latitudes and longitudes (Y and X), one of the upp refactoring tasks is to support ufs-weather-model inline post 2D decomposition functionality.

A ufs-weather-model PR was created for this test with inline post turned on in the control_2dwrtdecomp and regional_control_2dwrtdecomp tests,
https://github.com/ufs-community/ufs-weather-model/pull/1211

The upp/post_2d_decomp branch can be found at
https://github.com/WenMeng-NOAA/UPP/tree/post_2d_decomp

(Instructions: this, and all subsequent sections of text should be removed and filled in as appropriate.)
Provide a detailed description of what this PR does.  
What bug does it fix, or what feature does it add?

Several io/ subroutines are modified to call upp/post_2d_decomp functions
   
Is a change of answers expected from this PR?  

This PR does not change the output.


### Issue(s) addressed

https://github.com/ufs-community/ufs-weather-model/issues/1078

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)

## Testing

How were these changes tested?

1. Build the BASE with ufs-weather-model/develop
    run control fv3 and regional_control fv3  

2. Checkout the branch for this PR
    https://github.com/JesseMeng-NOAA/fv3atm/tree/feature/upp_unify_inline_2d
    run control_2dwrtdecomp and regional_control_2dwrtdecomp

3. Compare the results to confirm identical outputs are reproduced.


What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

https://github.com/WenMeng-NOAA/UPP/tree/post_2d_decomp
